extensions/positron-r/src/provider.ts

```diff
@@ -401,22 +441,34 @@ export async function findCurrentRBinary(): Promise<string | undefined> {
 		}
 	}
 
+	// TODO: for macOS, this should arguably be whatever
+	// /Library/Frameworks/R.framework/Versions/Current/ resolves to
+	// that would remove overlap between `findCurrentBinary()` and `findRBinaryFromPATH()`
+
 	cachedRBinary = await findRBinaryFromPATH();
 	return cachedRBinary;
 }
 
+let cachedRBinaryFromPATH: string | undefined;
```

Comment:
This is an unrelated improvement. Part of continuing effort to reduce rediscovery and repetition in the output channel.